Homework_36.ts

```diff
@@ -0,0 +1,169 @@
+					imdbRating: 8.6,
+					imdbVotes: 1259440,
+					type: "movie",
+					boxOffice: "$460,998,507",
+					production: "Lucasfilm Ltd.",
+				},
+				{ 
+					id: 4,
+					title: "Harry Potter and the Half-Blood Prince",
+					year: 2009,
+					released: "15 Jul 2009",
+					runtime: "153 min",
+					genre: ["Action", "Adventure", "Family"],
+					director: "David Yates",
+					writer: "Steve Kloves, J.K. Rowling",
+					actors: ["Daniel Radcliffe", "Emma Watson", "Rupert Grint"],
+					plot: "As Harry Potter begins his sixth year at Hogwarts, he discovers an old book marked as 'the property of the Half-Blood Prince' and begins to learn more about Lord Voldemort\'s dark past.",
+					country: "United Kingdom",
+					poster: "https://m.media-amazon.com/images/M/MV5BNzU3NDg4NTAyNV5BMl5BanBnXkFtZTcwOTg2ODg1Mg@@._V1_SX300.jpg",
+					imdbRating: 7.6,
+					imdbVotes: 492245,
+					boxOffice: "$302,305,431",
+					production: "Heyday Films, Warner Bros.",
+				}, 
+				{
+                    id: 5,
+					title: "Harry Potter and the Sorcerer's Stone",
+					year: 2001,
+					released: "16 Nov 2001",
+					runtime: "152 min",
+					genre: ["Adventure", "Family", "Fantasy"],
+					director: "Chris Columbus",
+					writer: "J.K. Rowling, Steve Kloves",
+					actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+					plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+					country: "United Kingdom, United States",
+					poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+					imdbRating: 7.6,
+					imdbVotes: 684604,
+					boxOffice: "$318,087,620",
+					production: "1492 Pictures, Heyday Films, Warner Brothers",
+				}
+    ]
+
+    //1. Собрать в массив все жанры фильмов (без повторения)
+
+    const getGenre = (array: Movies[]) => {
+        const genre = array.map(movie=> movie.genre)
+        const arrayGenres = []
+        return Array.from(new Set(arrayGenres.concat(...genre)))
+    }
+    console.log('All genres', getGenre(movies))
+
+    // 2. Собрать в массив всех актеров всех фильмов (без повторения)
+        
+    const getActors = (array: Movies[]) => {
+        const actors = array.map(movie=> movie.actors)
+        const arrayActors = []
+        return Array.from(new Set(arrayActors.concat(...actors)))
+    }
+    console.log('All actors', getActors(movies))
+
+    //3. Отсортировать фильмы по рейтингу по убыванию
+
+    const sortRating = (array: Movies[]) =>array.sort((a, b)=> b.imdbRating - a.imdbRating)
+    console.log('sorted by rating', sortRating(movies))
+
+    //4. Создать новый массив, где объекты фильмов будут состоять из следующих полей: id, title, released, plot
+
+    const getNewArrayMovies = (array: Movies[]) =>array.map(movie=>({id: movie.id, title: movie.title, released: movie.released, plot: movie.plot}))
```

Comment:
super!
my favorite way to do it
`movies.map(({id, title, released, plot})=>({id, title, released, plot}))`